RecoTauTag/RecoTau/interface/PFRecoTauClusterVariables.h

```diff
@@ -13,225 +16,38 @@
-    float tau_pt_weighted_deta_strip(const reco::PFTau& tau, int dm) const {
-      return pt_weighted_dx(tau, dm==10 ? 2 : 1, 1, dm);
-    }
-    float tau_pt_weighted_deta_strip(const pat::Tau& tau, int dm) const {
-      return pt_weighted_dx(tau, dm==10 ? 2 : 1, 1, dm);
-    }
-    /// return sum of pt weighted values of dphi relative to tau candidate for all pf photon candidates,
-    /// which are associated to signal
-    float tau_pt_weighted_dphi_strip(const reco::PFTau& tau, int dm) const {
-      return pt_weighted_dx(tau, dm==10 ? 2 : 1, 2, dm);
-    }
-    float tau_pt_weighted_dphi_strip(const pat::Tau& tau, int dm) const {
-      return pt_weighted_dx(tau, dm==10 ? 2 : 1, 2, dm);
-    }  
-    /// return sum of pt weighted values of dr relative to tau candidate for all pf photon candidates,
-    /// which are inside an isolation conde but not associated to signal
-    float tau_pt_weighted_dr_iso(const reco::PFTau& tau, int dm) const {
-      return pt_weighted_dx(tau, 2, 0, dm);
-    }
-    float tau_pt_weighted_dr_iso(const pat::Tau& tau, int dm) const {
-      return pt_weighted_dx(tau, 2, 0, dm);
-    }
-    /// return total number of pf photon candidates with pT>500 MeV, which are associated to signal
-    unsigned int tau_n_photons_total(const reco::PFTau& tau) const {
-      unsigned int n_photons = 0;
-      for (auto& cand : tau.signalPFGammaCands()) {
-        if ((float)cand->pt() > 0.5)
-          ++n_photons;
-      }
-      for (auto& cand : tau.isolationPFGammaCands()) {
-        if ((float)cand->pt() > 0.5)
-          ++n_photons;
-      }
-      return n_photons;
-    }
-    unsigned int tau_n_photons_total(const pat::Tau& tau) const {
-      unsigned int n_photons = 0;
-      for (auto& cand : tau.signalGammaCands()) {
-        if (cand->pt() > 0.5) 
-          ++n_photons;
-      }
-      for (auto& cand : tau.isolationGammaCands()) {  
-        if (cand->pt() > 0.5)
-	  ++n_photons;
-      }
-      return n_photons;
-    }
+namespace reco { namespace tau { namespace mva {
+  /// return chi2 of the leading track ==> deprecated? <==
+  float tau_leadTrackChi2(const reco::PFTau& tau);
+  /// return ratio of energy in ECAL over sum of energy in ECAL and HCAL
+  float tau_Eratio(const reco::PFTau& tau);
+  float tau_Eratio(const pat::Tau& tau);
+  /// return sum of pt weighted values of distance to tau candidate for all pf photon candidates, 
+  /// which are associated to signal; depending on var the distance is in 0=:dr, 1=:deta, 2=:dphi 
+  float pt_weighted_dx(const reco::PFTau& tau, int mode = 0, int var = 0, int decaymode = -1);
+  float pt_weighted_dx(const pat::Tau& tau, int mode = 0, int var = 0, int decaymode = -1);
+  /// return sum of pt weighted values of dr relative to tau candidate for all pf photon candidates,
+  /// which are associated to signal
+  float tau_pt_weighted_dr_signal(const reco::PFTau& tau, int dm);
+  float tau_pt_weighted_dr_signal(const pat::Tau& tau, int dm);
```

Comment:
these could stay inlined (perhaps with an explicit ```inline ``` qualifier).
Same for ```tau_pt_weighted_deta_strip```, ```tau_pt_weighted_dphi_strip```, and ```tau_pt_weighted_dr_iso```.